libkatana_python_native/include/katana/python/NumbaSupport.h

```diff
@@ -206,6 +263,24 @@ DefWithNumba(
       member_func>(cls, name, extra...);
 }
 
+template <auto func, typename Cls, typename... Extra, typename... ClsExtra>
+constexpr void
+DefWithNumba(
+    pybind11::class_<Cls, ClsExtra...>& cls,
+    std::enable_if_t<
+        std::is_same_v<
+            typename detail::StaticFunctionBinder<func>::StaticFunctionPointer,
+            decltype(func)>,
+        const char>* name,
+    const Extra&... extra) {
+  static_assert(
```

Comment:
Here and above, why do you use `static_assert` with the same condition as inside `std::enable_if_t`?